File: backend/EventsBackend/src/main/java/com/example/eventsbackend/model/Event.java
package com.example.eventsbackend.model;

public class Event {
    private String title;
    private String date;
    private String location;
    private String imageUrl;
    private String price;
    private String buyUrl;
    private String source;
    private String genre;

    private String description;

    public Event(){}

    public Event (String title, String date, String location, String imageUrl, String price, String buyUrl, String source, String genre, String description){
        this. title = title;
        this.date = date;
        this.location = location;
        this. imageUrl = imageUrl;
        this.price = price;
        this.buyUrl = buyUrl;
        this.source = source;
        this.genre = genre;
        this.description = description;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getPrice() {
        return price;
    }

    public void setPrice(String price) {
        this.price = price;
    }

    public String getBuyUrl() {
        return buyUrl;
    }

    public void setBuyUrl(String buyUrl) {
        this.buyUrl = buyUrl;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getGenre() {
        return genre;
    }
    public void setGenre(String genre) {
        this.genre = genre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}

File: frontend/events_application/lib/models/event.dart
class Event {
  final String title;
  final String date;
  final String location;
  final String imageUrl;
  final String price;
  final String buyUrl;
  final String source;
  final String genre;
  final String description;
  final String eventId;

  Event({
    required this.title,
    required this.date,
    required this.location,
    required this.imageUrl,
    required this.price,
    required this.buyUrl,
    required this.source,
    required this.genre,
    required this.description,
    required this.eventId,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      title: json['title'] ?? '',
      date: json['date'] ?? '',
      location: json['location'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      price: json['price'] ?? '',
      buyUrl: json['buyUrl'] ?? '',
      source: json['source'] ?? '',
      genre: json['genre'] ?? '',
      description: json['description'] ?? '',
      eventId: json['eventId'].toString(),
    );
  }
}
